src/lit/youtube-modal.ts

```diff
@@ -0,0 +1,284 @@
+ *
+ * window.customElements.define('degu-youtube-inline', DeguYoutubeInline);
+ * window.customElements.define('degu-youtube-modal', DeguYoutubeModal);
+ *
+ * // Define a host.  When you register against a host, the component will
+ * // look for any clicks within that host with the data attribute
+ * // `youtube-modal="<youtube-id>".
+ * const myHost = document.body;
+ * DeguYoutubeModal.register(myHost)
+ * ```
+ *
+ * Now somewhere within your host, just add buttons or elements with the
+ * attribute `youtube-modal`.
+ *
+ * ```html
+ *   <button youtube-modal="8Qn_spdM5Zg">Watch StarWars</button>
+ *   <button youtube-modal="TcMBFSGVi1c">Watch Avengers</button>
+ * ```
+ *
+ */
+export class DeguYoutubeModal extends LitElement {
+  private watcher: DomWatcher;
+  private hasFirstUpdateCompleted = false;
+  private isOpen = false;
+
+  private host: HTMLElement;
+  private delegateDisposer: Function = null;
+
+  @query('summary')
+  summary: HTMLElement;
+
+  @query('details-dialog')
+  detailsDialog: DetailsDialogElement;
+
+  @query('details')
+  details: HTMLDetailsElement;
+
+  @query('degu-youtube-inline')
+  youtubeInline: DeguYoutubeInline;
+
+  private youtubeId: string;
+
+  constructor(host: HTMLElement) {
+    super();
+    this.host = host;
+    this.delegateDisposer = dom.addDelegatedListener(
+      this.host,
+      'click',
+      (el: HTMLElement) => {
+        const youtubeModalId = el.getAttribute('youtube-modal');
+        if (youtubeModalId) {
+          // Attach this element if it is not in the dom yet.
+          if (!this.isConnected) {
+            this.host.appendChild(this);
+          }
+
+          // Wait for child components to render.
+          window.setTimeout(() => {
+            this.openModalAndPlay(youtubeModalId);
+          });
+        }
+      }
+    );
+  }
+
+  connectedCallback() {
+    super.connectedCallback();
+    this.watcher = new DomWatcher();
+
+    this.updateComplete.then(this.onUpdateComplete.bind(this));
+  }
+
+  private onUpdateComplete() {
+    if (!this.hasFirstUpdateCompleted) {
+      // Watch and update the toggle state of the modal.
+      this.watcher.add({
+        element: this.details,
+        on: 'toggle',
+        callback: this.onToggle.bind(this),
+      });
+      this.hasFirstUpdateCompleted = true;
+    }
+  }
+
+  openModalAndPlay(youtubeId: string) {
+    this.youtubeId = youtubeId;
+    this.detailsDialog.toggle(true);
+  }
+
+  private onToggle() {
+    this.isOpen = this.details.open;
+    if (this.isOpen && this.youtubeInline) {
+      this.youtubeInline.load(this.youtubeId, true);
+    } else {
+      this.youtubeInline.pause();
+    }
+  }
+
+  toggle(open: boolean) {
+    this.detailsDialog.toggle(open);
+  }
+
+  /**
+   * Creates an instance of YoutubeModal.
+   */
+  static register(host: HTMLElement): DeguYoutubeModal {
+    const modal = new DeguYoutubeModal(host);
+    return modal;
+  }
+
+  static styles = css`
+    .modal summary {
```

Comment:
No concerns here; can you test it in Safari to make sure the "details marker" doesn't show up? If it does, we need to add a CSS override to remove it.

Reply:
I tested it on Safari and it looked the same so I think we are good here.